### PR TITLE
Added license for libreJS to identify, fixes #543

### LINKED ISSUE #543: Support LibreJS
Currently, when LibreJS is enabled, it blocks javascript loaded from a plinth instance.

Since all of the JS used by Plinth is free software, we should be able to add licensing information so that the scripts won't be blocked. There is a guide on how to add license info in a way that LibreJS can recognize:
https://www.gnu.org/software/librejs/free-your-javascript.html

### PR DESCRIPTION
**Pull request by 500swapnil**

### CODE REVIEW

File: plinth/templates/base.html [hunk 1]
```diff
@@ -222,5 +222,6 @@
 
   {% block app_js %}<!-- placeholder for app-specific js files -->{% endblock %}
   {% block page_js %}<!-- placeholder for page-specific js files -->{% endblock %}
+  <a href="/static/jslicense.html" data-jslicense="1">JavaScript license information</a>
```
Comment (jvalleroy) [inline plinth/templates/base.html]: It looks like this should be /plinth/static/jslicense.html, that's the page that's working for me.

Comment (500swapnil) [inline plinth/templates/base.html]: But there's no static folder in plinth.

Comment (500swapnil) [inline plinth/templates/base.html]: static folder is in the main folder itself

Comment (jvalleroy) [inline plinth/templates/base.html]: Ah we should use the static tag, something like this:
```
href="{% static 'jslicense.html' %}"
```
Then Django should pick the correct folder where static files are being provided.

Comment (500swapnil) [inline plinth/templates/base.html]: Oh ok

Comment (500swapnil) [inline plinth/templates/base.html]: Is it ok now?

File: static/jslicense.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	<style>a {
+		text-align:center;
+		width:300;
+	}
+	</style>
+<body>
+<h1>JavaScript Web labels table</h1>
+<table id="jslicense-labels1" style="text-align:center; width:500">
+    <tr>
+        <td><a href="/plinth/modules/xmpp/static/jsxc-plinth.js">jsxc-plinth.js</a></td>
```
Comment (jvalleroy) [inline static/jslicense.html]: This link doesn't work for me.
The one that works is: /plinth/static/xmpp/jsxc-plinth.js

File: plinth/templates/base.html [hunk 3]
```diff
@@ -222,5 +222,6 @@
 
   {% block app_js %}<!-- placeholder for app-specific js files -->{% endblock %}
   {% block page_js %}<!-- placeholder for page-specific js files -->{% endblock %}
+  <a href="{% static 'jslicense.html' %}" data-jslicense="1">JavaScript license information</a>
```
Comment (jvalleroy) [inline plinth/templates/base.html]: I suggest to move this inside the footer_block, and center the text. That should improve the appearance a bit.

File: static/jslicense.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	<style>a {
+		text-align:center;
+		width:300;
+	}
+	</style>
+<body>
+<h1>JavaScript Web labels table</h1>
+<table id="jslicense-labels1" style="text-align:center; width:500">
+    <tr>
+        <td><a href="/plinth/static/jsxc-plinth.js">jsxc-plinth.js</a></td>
```
Comment (jvalleroy) [inline static/jslicense.html]: Just missing the /xmpp part I think...(/plinth/static/xmpp/jsxc-plinth.js)

**Comment by jvalleroy** [comment]
@500swapnil We also have some inline JS in a few templates:
- https://github.com/freedombox/Plinth/blob/master/plinth/modules/xmpp/templates/jsxc.html#L72
- https://github.com/freedombox/Plinth/blob/master/plinth/modules/networks/templates/connections_edit.html#L39
- https://github.com/freedombox/Plinth/blob/master/plinth/modules/pagekite/templates/pagekite_configure.html#L48
- https://github.com/freedombox/Plinth/blob/master/plinth/modules/tor/templates/tor.html#L143
- https://github.com/freedombox/Plinth/blob/master/plinth/modules/dynamicdns/templates/dynamicdns_configure.html#L46

Maybe we can add the ```@license``` and ```@license-end``` comments around these.

**Comment by 500swapnil** [comment]
Yeah I can do that! Thanks!

**Comment by jvalleroy** [comment]
@500swapnil  I don't think that's the correct license though. Other than the JSXC script, I believe everything else is licensed AGPL-3.0 or later.

Also, I think we can use the magnet link, rather then include the whole license text.

**Comment by 500swapnil** [comment]
Ok. I will add the magnet link for all except jscx.html ok?

**Comment by 500swapnil** [comment]
I am adding the magnet link in jsxc as well

**Comment by jvalleroy** [comment]
@500swapnil Thanks, I've merged these changes, with minor updates.

This doesn't completely get LibreJS working with Plinth though. I'll open a new issue with my findings.

**Comment by 500swapnil** [comment]
Thanks! Any other issue I can help with?

**Comment by jvalleroy** [comment]
@500swapnil I opened #691 for remaining issues related to LibreJS.

**Comment by 500swapnil** [comment]
@jvalleroy Okay I will have a look. Thank you! But are there any other issues other than libreJS I can help with?

**Comment by jvalleroy** [comment]
@500swapnil Sure, do any issues in this list look interesting to you?
https://github.com/freedombox/Plinth/issues?q=is%3Aopen+is%3Aissue+label%3Abeginner
In particular, #322, #316, #254, or #220, those are the ones I think I'm somewhat familiar with what the solution might look like.

**Comment by 500swapnil** [comment]
Thank you! I will have a look at them

**Pull request closed without merging**